### PR TITLE
Remove Publish Later task

### PR DESCRIPTION
**Pull request by GiuseppeMusso-Laser**
Publish Later module: added button "Remove Publish Later" to cancel publish later task.
This PR should partially solve issue #860.

### CODE REVIEW

**Comment by sebastienros** [comment]
There is a breaking change here.

Also, can anyone find the issue where we talked about changing the UI for this feature?

**Comment by HermesSbicego-Laser** [comment]
@sebastienros, is it this one #7056? if yes, someone is working on that? Our report was intended to have a way to cancel the publish later Task (there is no way to do that by now), so it's a functional proposition not only a UI refactoring. 

Also, i don't really like what is proposed there, because it's difficult to understand what the `Publish` button should do when the "Publish later" is checked: a `Publish later` or a `Publish now`? And also in order to cancel the `Publish later Task` what the user should do? It should to remove the flag and press `Save`? 

Thanks

File: src/Orchard.Web/Modules/Orchard.PublishLater/Services/IPublishLaterService.cs [hunk 1]
```diff
@@ -6,5 +6,6 @@ namespace Orchard.PublishLater.Services {
     public interface IPublishLaterService : IDependency {
         DateTime? GetScheduledPublishUtc(PublishLaterPart publishLaterPart);
         void Publish(ContentItem contentItem, DateTime scheduledPublishUtc);
+        void RemovePublishLaterTasks(ContentItem contentItem);
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.PublishLater/Services/IPublishLaterService.cs]: Remove it.

File: src/Orchard.Web/Modules/Orchard.PublishLater/Services/PublishLaterService.cs [hunk 2]
```diff
@@ -31,5 +31,9 @@ public class PublishLaterService : IPublishLaterService {
             IScheduledTask task = _publishingTaskManager.GetPublishTask(publishLaterPart.ContentItem);
             return (task == null ? null : task.ScheduledUtc);
         }
+
+        public void RemovePublishLaterTasks(ContentItem contentItem) {
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.PublishLater/Services/PublishLaterService.cs]: Delete this

File: src/Orchard.Web/Modules/Orchard.PublishLater/Drivers/PublishLaterPartDriver.cs [hunk 3]
```diff
@@ -105,6 +105,9 @@ public class PublishLaterPartDriver : ContentPartDriver<PublishLaterPart> {
                 }
             }
 
+            if (httpContext.Request.Form["submit.Save"] == "submit.RemovePublishLaterTasks") {
+                _publishLaterService.RemovePublishLaterTasks(model.ContentItem);
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.PublishLater/Drivers/PublishLaterPartDriver.cs]: Use the `IPublishTaskManager` service directly

**Comment by sebastienros** [comment]
Check if the fields are empty when versioning, if so, remove any task. We don't add a button right now, we'll find a better UI later.

**Pull request merged**